### DISCUSSION TITLE
Is there any way to upload and serve images in the nextjs application in runtime?

### DISCUSSION THREAD

**Discussion by tejassharma-ts** [vercel/next.js#62129] [Help] [answered]
### Summary

I am developing a web application that includes a feature to upload images to the file system and serve them to the frontend. In the development environment, placing images in the 'public' directory works as expected. However, in the production environment, it is not functioning properly. The documentation recommends using 'Vercel Blob,' or alternatively, I could use Amazon S3. However, I am wondering if there is a way to achieve this without relying on an external service.

Thank you

**Comment by icyJoseph** [comment, marked as answer]
I am afraid not:

- https://github.com/orgs/vercel/discussions/3582#discussioncomment-6732865

**Reply by tejassharma-ts** [reply]
Thanks, will go for other service for that than.

**Reply by tejassharma-ts** [reply]
Hey! Sorry, can we set up an Nginx server and do something with it to solve the issue like uploading images to the filesystem. Nginx server will then serve that image for us when requested.
